samples/nrf9160/gnss/src/main.c

```diff
@@ -70,6 +81,18 @@ static void gnss_event_handler(int event)
 		}
 		break;
 
+#if defined(CONFIG_GNSS_SAMPLE_MODE_TTFF_TEST)
+	case NRF_MODEM_GNSS_EVT_FIX:
+		/* Time to fix is calculated here, but it's printed from a delayed work to avoid
+		 * messing up the NMEA output.
+		 */
+		time_to_fix = (k_uptime_get() - fix_timestamp) / 1000;
+		k_work_schedule_for_queue(&gnss_work_q, &ttff_test_got_fix_work, K_MSEC(100));
+		k_work_schedule_for_queue(&gnss_work_q, &ttff_test_prepare_work,
+					  K_SECONDS(CONFIG_GNSS_SAMPLE_MODE_TTFF_TEST_INTERVAL));
+		break;
```

Comment:
I agree, that would be nice. It would make it more visible to the user that the actual time GNSS is running is sometimes shorter than the printed TTFF. I will have a look how it could be done. One problem is, that LTE might block GNSS in the beginning, then GNSS might get time to run but be blocked again before a fix is acquired.

One simple option could be to calculate PVT notifications which have the "deadline missed" bit set. PVT notifications come once a second, so we can estimate that the time GNSS has been blocked in seconds is the same as the number of PVT notifications with the bit set.

Reply:
I implemented it like a suggested above. Seems to work as expected.

Reply:
That's a neat way of doing it